### ISSUE TITLE
Records repeated with `order by @rid desc`

### ISSUE THREAD

**Issue by gongdo** [orientechnologies/orientdb#5298]
I'm using v2.1.5, and this can be found both Studio and Console.

When I query this on GratefulDeadConcerts database,
```
SELECT FROM V WHERE @rid < #9:5
```
I get 5 records which @rid is less than #9:5 as expected.

But with this,
```
SELECT FROM V WHERE @rid < #9:5 ORDER BY @rid DESC
```
I get repeated records from #9:4 to #9:0 like this:
```
----+----+------+
#   |@RID|@CLASS|
----+----+------+
0   |#9:4|V     |
1   |#9:3|V     |
2   |#9:2|V     |
3   |#9:1|V     |
4   |#9:0|V     |
5   |#9:4|V     |
6   |#9:3|V     |
7   |#9:2|V     |
8   |#9:1|V     |
9   |#9:0|V     |
10  |#9:4|V     |
11  |#9:3|V     |
12  |#9:2|V     |
13  |#9:1|V     |
14  |#9:0|V     |
15  |#9:4|V     |
16  |#9:3|V     |
17  |#9:2|V     |
18  |#9:1|V     |
19  |#9:0|V     |
----+----+------+
LIMIT EXCEEDED: resultset contains more items not displayed (limit=20)
```
It repeats as much as the total records reach to the limit.

It looks like a bug. Can I get a workaround for this?

**Comment by smolinari** [comment]
I can confirm the bug happening within Studio and 2.1.5 (Enterprise).

Scott

**Comment by luigidellaquila** [comment]
Just fixed in 2.1.x branch. 
The problem is not reproducible in develop branch.
@gongdo please check it let me know if I can close

Thanks

Luigi

**Comment by gongdo** [comment]
I've just checked 2.1.x version, and it works well, even with order by, skip and limit combination.
Thanks!
